ios/RCTWebRTC/SocketConnection.m

```diff
@@ -0,0 +1,173 @@
+    }
+    
+    if (![self setupSocketBufferSize]) {
+        return  nil;
+    }
+    
+    if (![[NSFileManager defaultManager] createFileAtPath:filePath contents:nil attributes:nil]) {
+        NSLog(@"failure creating socket file");
+        return nil;
+    }
+    
+    if (![self setupSocketWithFileAtPath: filePath]) {
+        return nil;
+    }
+    
+    return self;
+}
+
+- (void)openWithStreamDelegate:(id <NSStreamDelegate>)streamDelegate {
+    int status = listen(self.serverSocket, 10);
+    if (status < 0) {
+        NSLog(@"failure: socket listening");
+        return;
+    }
+    
+    dispatch_source_t listeningSource = dispatch_source_create(DISPATCH_SOURCE_TYPE_READ, self.serverSocket, 0, NULL);
+    dispatch_source_set_event_handler(listeningSource, ^ {
+        int clientSocket = accept(self.serverSocket, NULL, NULL);
+        if (clientSocket < 0) {
+            NSLog(@"failure accepting connection");
+            return;
+        }
+                
+        CFReadStreamRef readStream;
+        CFWriteStreamRef writeStream;
+        
+        CFStreamCreatePairWithSocket(kCFAllocatorDefault, clientSocket, &readStream, &writeStream);
+
+        self.inputStream = (__bridge_transfer NSInputStream *)readStream;
+        self.inputStream.delegate = streamDelegate;
+        [self.inputStream setProperty:@"kCFBooleanTrue" forKey:@"kCFStreamPropertyShouldCloseNativeSocket"];
+        
+        self.outputStream = (__bridge_transfer NSOutputStream *)writeStream;
+        [self.outputStream setProperty:@"kCFBooleanTrue" forKey:@"kCFStreamPropertyShouldCloseNativeSocket"];
+        
+        [self.networkThread start];
+        [self performSelector:@selector(scheduleStreams) onThread:self.networkThread withObject:nil waitUntilDone:true];
+                
+        [self.inputStream open];
+        [self.outputStream open];
+    });
+    
+    self.listeningSource = listeningSource;
+    dispatch_resume(listeningSource);
+}
+
+- (void)close {    
+    [self performSelector:@selector(unscheduleStreams) onThread:self.networkThread withObject:nil waitUntilDone:true];
+    
+    self.inputStream.delegate = nil;
+    self.outputStream.delegate = nil;
+        
+    [self.inputStream close];
+    [self.outputStream close];
+        
+    [self.networkThread cancel];
+    
+    close(self.serverSocket);
+    dispatch_source_cancel(self.listeningSource);
+}
+
+// MARK: - Private Methods
+
+- (void)setupNetworkThread {
+    self.networkThread = [[NSThread alloc] initWithBlock:^{
+        do {
+            @autoreleasepool {
+                [[NSRunLoop currentRunLoop] run];
+            }
+        } while (![NSThread currentThread].isCancelled);
+    }];
+    self.networkThread.qualityOfService = NSQualityOfServiceUserInitiated;
+}
+
+- (BOOL)setupSocketBufferSize {
+    int optval = kConnectionBufferSize;
+    socklen_t optlen = sizeof(optval);
+    
+    int status = setsockopt(self.serverSocket, SOL_SOCKET, SO_SNDBUF, &optval, optlen);
```

Comment:
This probably has no effect at the moment. It needs to run on the connected socket, so in what you get after calling accept(), not in the server socket itself.

Reply:
Fixed. Updated to run on the `clientSocket`